model/legacy-private/src/main/java/org/keycloak/storage/UserStorageManager.java

```diff
@@ -484,11 +484,20 @@ public int getUsersCount(RealmModel realm, Map<String, String> params, Set<Strin
     @Override
     public Stream<UserModel> searchForUserStream(RealmModel realm, Map<String, String> attributes, Integer firstResult, Integer maxResults) {
         if(attributes.containsKey(CustomSearchKey.LDAP_PROVIDER_ID)) {
-        	Stream<UserModel> results = queryByProviderId((provider, firstResultInQuery, maxResultsInQuery) -> {
-        		return ((UserQueryMethodsProvider)provider).searchForUserStream(realm, attributes,
-    					firstResultInQuery, maxResultsInQuery);
-        	}, realm, firstResult, maxResults, attributes.get(CustomSearchKey.LDAP_PROVIDER_ID));
-        	return results;
+        	if(attributes.get(CustomSearchKey.LDAP_PROVIDER_ID) == null || attributes.get(CustomSearchKey.LDAP_PROVIDER_ID).equalsIgnoreCase("null")) {
```

Comment:
@AnkurJaiswal1990  We can use StringUtils.hasText for null check

Reply:
@thakkarsmruti Library is not available